### ISSUE TITLE
Compatibility issue with Vue Apollo

### ISSUE THREAD

**Issue by harryjubb** [emedware/vue-golden-layout#77]
Hi,

I'm having some trouble getting `vue-golden-layout` working with `vue-apollo`.

Using either on their own works without an issue, but having both `Vue.use(vgl)` and `Vue.use(VueApollo)` causes console errors and the `<golden-layout></golden-layout>` to not appear. I've also tried importing the `goldenLayout` component instead of using `Vue.use` [as far as I can tell, the plugin install is only registering the components(?)] and get the same issue.

Console log:

![image](https://user-images.githubusercontent.com/4588417/85839640-1cd4b900-b793-11ea-88da-7a401d34f5f8.png)

I've put together a minimal reproducible example: https://github.com/harryjubb/apollo-gl-test see `src/main.ts` and `src/App.vue`

As far as I can tell, it seems to be an issue merging the `data` properties from `vue-apollo` and `vue-golden-layout`, but I'm not familiar enough with Vue internals to say for sure what's going on.

Any help or direction towards fixing it would be much appreciated.

**Comment by harryjubb** [comment]
I believe I've found the issue:

Having has a look through the `vue-golden-layout` source, it looks like there is a `data` `mixin` that doesn't return anything. 

Presumably that results in that component having an `undefined` `data` property, which Vue then can't merge with `data` from `vue-apollo`.

Returning a `{}` from that `mixin` `data` function removes the console errors, and `vue-golden-layout` works again 🎉 

See: https://github.com/harryjubb/vue-golden-layout/commit/e9f6986dcb5b8c94f9c598f1b11c8cc34c98ed4e

I tested this on a branch of my MRE https://github.com/harryjubb/apollo-gl-test/tree/vue-apollo-compat

I'm not sure if returning a `{}` here would have any negative effects? If not, I'd be happy to PR this (or for it just to be added, as it's a one-line fix!)
